Search.setIndex({"docnames": ["Case Study", "Installation", "TRIAGEcluster", "TRIAGEgene", "TRIAGEparser", "Testing TRIAGE", "User-friendly functions", "index"], "filenames": ["Case Study.rst", "Installation.rst", "TRIAGEcluster.rst", "TRIAGEgene.rst", "TRIAGEparser.rst", "Testing TRIAGE.rst", "User-friendly functions.rst", "index.rst"], "titles": ["Case Study", "Installation", "TRIAGEcluster", "TRIAGEgene", "TRIAGEparser", "Testing TRIAGE", "User-friendly Functions", "Welcome to TRIAGE R Package\u2019s documentation!"], "terms": {"thi": [0, 1, 2, 3, 4, 5, 6, 7], "demonstr": [0, 5], "applic": [0, 3], "triag": [0, 2, 3, 4, 6], "analysi": [0, 1, 2, 3, 4, 5, 6, 7], "pipelin": 0, "us": [0, 1, 2, 3, 4, 5, 6, 7], "publicli": 0, "avail": 0, "singl": [0, 2, 3, 4, 7], "nucleu": 0, "rna": 0, "sequenc": [0, 2], "snrna": 0, "seq": [0, 2, 3, 4, 5], "dataset": [0, 3, 5], "data": [0, 1, 2, 3, 4, 5, 6, 7], "sourc": [0, 1, 3], "kupp": 0, "et": [0, 2, 3, 4], "al": [0, 2, 3, 4], "natur": [0, 3], "2022": 0, "spatial": 0, "multi": 0, "omic": 0, "map": [0, 3, 4], "human": [0, 1, 3, 5, 6], "myocardi": 0, "infarct": 0, "http": [0, 1, 3, 6], "www": [0, 3], "com": [0, 1], "articl": 0, "s41586": 0, "022": 0, "05060": 0, "x": 0, "The": [0, 1, 2, 3, 4, 5, 6], "control": 0, "endotheli": [0, 6], "pericyt": 0, "cell": [0, 2, 3, 4, 5, 6, 7], "showcas": 0, "object": [0, 5, 6], "To": [0, 1, 5], "appli": [0, 3, 5], "triageclust": [0, 7], "cluster": [0, 2, 3, 4, 5, 6], "bypeak": [0, 7], "calcul": [0, 3, 4, 5, 6], "averag": [0, 5, 6], "express": [0, 2, 3, 4, 5, 6], "variou": [0, 2], "categori": 0, "triagegen": [0, 6, 7], "gener": [0, 2, 3, 5, 6], "weight": [0, 2, 3, 4, 5, 6], "plotjaccard": [0, 7], "jaccard": [0, 5, 6], "index": [0, 5, 6, 7], "heatmap": [0, 5, 6], "visual": [0, 2, 5, 6], "triagepars": [0, 6, 7], "dissect": 0, "gene": [0, 1, 2, 3, 4, 5, 6], "associ": [0, 4, 5], "ontologi": [0, 4, 5], "plotgo": [0, 7], "go": [0, 1, 4, 5, 6], "enrich": [0, 1, 4, 5, 6], "step": [0, 1, 5, 7], "load": [0, 1], "packag": [0, 2, 3, 4, 5, 6], "set": [0, 1, 2, 3, 4, 6], "up": [0, 1], "work": [0, 1, 5, 6], "directori": [0, 2, 4, 5, 6], "librari": [0, 1, 5], "reticul": [0, 1, 5], "setwd": [0, 5], "path": [0, 1, 2, 4, 5, 6], "1": [0, 1, 2, 3, 4, 5, 6], "run": [0, 2, 5], "select": [0, 3, 5, 6], "suitabl": [0, 1, 4, 5], "bandwidth": [0, 2, 5], "resolut": [0, 2], "base": [0, 3, 5, 6], "umap": [0, 2, 5], "review": [0, 5], "function": [0, 1, 2, 3, 4, 5, 7], "expr_fil": [0, 5], "system": [0, 3, 5], "file": [0, 1, 2, 4, 5, 6], "extdata": [0, 5], "triagecluster_demo_expr_human": [0, 5], "csv": [0, 2, 4, 5, 6], "metadata_fil": [0, 5], "triagecluster_demo_metadata_human": [0, 5], "outdir": [0, 2, 4, 5, 6], "test": [0, 7], "output_prefix": [0, 2, 5], "demo": [0, 5], "peak_fil": [0, 5], "demo_bw0": [0, 5], "80_metadata": [0, 5], "avg_peak": [0, 5], "save": [0, 5, 6], "result": [0, 2, 3, 4, 5, 6], "write": [0, 5], "averagebypeak": [0, 5], "row": [0, 5, 6], "name": [0, 2, 4, 5, 6], "true": [0, 1, 3, 5], "quot": [0, 5], "fals": [0, 3, 4, 5], "nine": [0, 2], "from": [0, 1, 2, 3, 4, 5, 6], "0": [0, 2, 3, 4, 6], "9": [0, 1, 2], "correspond": [0, 2, 3], "metadata": [0, 2, 5, 6], "here": [0, 1, 3], "8": [0, 6], "wa": [0, 1], "downstream": [0, 6], "four": [0, 5], "defin": 0, "peak": [0, 2, 5, 6], "within": [0, 1, 2, 3], "A": [0, 2, 4, 6], "snippet": 0, "i": [0, 1, 2, 3, 4, 5, 6], "shown": [0, 2], "below": [0, 1], "peak0": [0, 5, 6], "peak1": [0, 6], "peak2": 0, "peak3": 0, "aldh1a2": 0, "04404541": 0, "04426352": 0, "18265890": 0, "12415153": 0, "prrx1": 0, "63263328": 0, "19289223": 0, "08957362": 0, "11108879": 0, "nova2": 0, "00785169": 0, "31212372": 0, "22311596": 0, "28179011": 0, "ngfr": 0, "07502976": 0, "01140611": 0, "01078702": 0, "02280367": 0, "mcf2l": 0, "03431053": 0, "41983845": 0, "25927713": 0, "39706381": 0, "tbx4": 0, "2": [0, 1, 2, 3, 4, 5, 6], "similar": [0, 3, 6], "d": [0, 2, 3, 5, 6], "averagebypeak_d": [0, 5], "tab": [0, 2, 4, 5], "delimit": [0, 2, 4, 5], "text": [0, 4, 5], "tabl": [0, 1, 4, 5], "txt": [0, 2, 4, 5, 6], "sep": [0, 5], "t": [0, 1, 5], "col": [0, 5], "na": [0, 5], "jaccard_heatmap_peak": 0, "pdf": [0, 5, 6], "000685619": 0, "000689014": 0, "018409486": 0, "001932566": 0, "019265802": 0, "005874214": 0, "002727817": 0, "003383026": 0, "000207595": 0, "008252406": 0, "005899082": 0, "0074504": 0, "012602077": 0, "001915782": 0, "001811799": 0, "003830128": 0, "000594645": 0, "024607587": 0, "004493599": 0, "024212873": 0, "extent": 0, "across": [0, 3, 4, 6], "identifi": [0, 2, 3, 6], "3": [0, 1, 2, 3, 4, 5, 6], "input_fil": [0, 5, 6], "input_typ": [0, 4, 5], "triageparser_peak": 0, "indir": [0, 5, 6], "output": [0, 5, 6, 7], "compris": 0, "which": [0, 1, 2, 3, 4, 6], "ar": [0, 1, 2, 4, 5, 6], "organ": [0, 1, 5, 6], "two": [0, 1, 2, 3, 4, 6], "separ": [0, 1], "folder": [0, 4], "gene_clust": [0, 4, 5, 6], "each": [0, 1, 2, 3, 4, 5, 6], "structur": 0, "illustr": 0, "altern": 0, "follow": [0, 1, 2], "avg_clust": 0, "peak_col": 0, "final_clust": 0, "averagebyclust": 0, "ds_cluster": 0, "averagebycluster_d": 0, "jaccard_heatmap_clust": 0, "triageparser_clust": 0, "type": [0, 1, 3, 4], "avg_celltyp": 0, "cell_typ": 0, "averagebycelltyp": 0, "ds_celltyp": 0, "averagebycelltype_d": 0, "jaccard_heatmap_celltyp": 0, "triageparser_celltyp": 0, "section": 1, "cover": [1, 6], "requir": [1, 2, 3, 4], "both": [1, 2], "pheatmap": 1, "modul": 1, "It": [1, 4, 6], "": [1, 3, 6], "recommend": [1, 2, 3], "manag": 1, "note": [1, 2, 3, 4], "ani": [1, 3, 4, 6], "version": [1, 6], "we": [0, 1, 2, 3], "5": [1, 2, 3, 5, 6], "an": [1, 4], "exampl": [1, 6, 7], "install_python": 1, "py_instal": 1, "panda": 1, "envnam": 1, "numpi": [], "scipi": 1, "matplotlib": 1, "request": 1, "scikit": 1, "learn": [1, 5], "seaborn": 1, "when": [1, 3, 4], "import": [], "understand": [], "how": [3, 5], "kei": [5, 6], "point": 4, "consid": [2, 4, 6], "intellig": [], "design": [4, 5, 6], "your": [], "specifi": [2, 3, 4, 5, 6], "check": [], "whether": 3, "alreadi": 1, "target": [], "No": 4, "redund": [], "If": [1, 3, 4], "given": 6, "attempt": [], "reinstal": [], "avoid": 1, "time": [], "especi": [], "first": 1, "new": 1, "session": 1, "you": [1, 4], "can": [1, 2, 3, 4, 5, 6], "where": 6, "want": 4, "done": [], "paramet": [6, 7], "For": [1, 2, 3, 4], "instanc": [], "In": [1, 3, 4], "futur": 1, "activ": 1, "use_virtualenv": 1, "download": 1, "e": [1, 2, 4, 6], "g": 1, "triage_1": 1, "tar": 1, "gz": 1, "github": 1, "repositori": 1, "qiongyi": 1, "triage_r_packag": 1, "repo": 1, "null": [1, 2, 3, 6], "one": [1, 2, 3, 4], "core": [2, 3, 4], "r": [2, 3, 4, 5, 6], "repres": [2, 3, 6], "notabl": 2, "advanc": 2, "scrna": [2, 5], "integr": [2, 3], "epigenom": [2, 3, 4], "signatur": [2, 3], "demarc": 2, "divers": [2, 4], "heterogen": 2, "util": [2, 3], "combin": 2, "high": [2, 6], "rt": [2, 3], "kernel": 2, "densiti": 2, "estim": 2, "dimension": 2, "space": [2, 4], "leverag": 2, "refin": [2, 5], "biolog": [2, 4, 5], "distinct": [1, 2, 3, 4, 5], "popul": 2, "refer": [2, 3], "our": [2, 3], "method": [2, 3, 5, 6], "more": [1, 2, 3, 4], "detail": [1, 2, 3, 4], "see": [1, 2, 3, 4, 6], "sun": [2, 4], "nucleic": [2, 4], "acid": [2, 4], "research": [2, 3, 4, 5, 6], "2023": [2, 4], "infer": [2, 3, 4], "consortium": [2, 4], "scale": [2, 4], "epigenet": [2, 3, 4], "anchor": [2, 4], "ident": [2, 3, 4], "normal": [2, 3], "matrix": [2, 3, 5, 6], "either": [2, 3, 4, 6], "format": [2, 3, 4, 5, 6], "need": [1, 2], "contain": [2, 4, 6], "inform": [2, 4], "about": 2, "coordin": 2, "also": [1, 2, 3, 6], "total": 2, "18": 2, "These": [1, 2, 5, 6], "includ": [1, 2, 3, 4, 5, 6], "plot": [2, 5, 6], "differ": [2, 4, 6], "rang": 2, "alongsid": 2, "characterist": [2, 3], "expr": [2, 6], "accept": [2, 3], "option": [1, 2, 3, 4, 6], "default": [2, 3, 4, 5, 6], "triagecluster_result": 2, "prefix": [2, 6], "cell_column": [2, 5, 6], "indic": [2, 3], "column": [2, 6], "barcod": [2, 5, 6], "umap_column": 2, "umap_": 2, "priority_rt": 2, "prioriti": 2, "list": [1, 2, 4, 5, 6], "priority_epimap_rt": 2, "intend": 2, "user": [2, 3, 7], "who": 2, "capabl": [2, 5], "own": 2, "most": [2, 3], "ensur": [2, 3, 5], "optim": [2, 4], "min_cells_per_peak": 2, "minimum": [2, 3], "number": [2, 3, 4, 6], "per": [2, 3], "seed": 2, "valu": [2, 3, 4, 5, 6], "reproduc": [2, 3], "discord": [2, 3, 6], "score": [2, 3, 5, 6], "project1": 2, "info": [2, 3], "umap1": 2, "umap2": 2, "123": 2, "project2": 2, "expr_matrix": 2, "project3": 2, "4": [0, 2, 3, 5, 6], "project4": 2, "aim": 3, "predict": 3, "regulatori": 3, "potenti": [1, 3], "further": [0, 3], "genet": 3, "driver": 3, "approach": [1, 3, 6], "repress": 3, "tendenc": 3, "analyz": [3, 4, 5], "broad": 3, "h3k27me3": [3, 4], "domain": 3, "5kb": 3, "upstream": 3, "plu": 3, "bodi": 3, "metric": 3, "previou": 3, "literatur": 3, "after": 3, "transform": [3, 5], "becom": 3, "instrument": 3, "rank": [3, 4, 5, 6], "them": 3, "descend": [3, 4, 6], "order": [3, 6], "group": [3, 4, 5, 6], "vari": 3, "specif": [1, 3, 4, 5, 6], "condit": 3, "individu": [3, 6], "sampl": [3, 4], "even": 3, "shim": 3, "2020": 3, "conserv": 3, "logic": 3, "govern": 3, "adapt": [3, 4, 5, 6], "protein": [3, 4], "code": [3, 4], "non": [3, 4], "rnaseq": [3, 4], "proteom": [3, 4], "chip": [3, 4], "frame": [3, 6], "like": 3, "count": 3, "million": 3, "cpm": 3, "fragment": 3, "kilobas": 3, "transcript": 3, "read": [3, 5, 6], "fpkm": 3, "tpm": 3, "help": 3, "crucial": 3, "m": 3, "among": 3, "other": [3, 4, 6], "speci": [1, 3, 5, 6], "c": [1, 2, 3, 5, 6], "intestinali": 3, "chicken": 3, "guinea": 3, "pig": 3, "mous": [3, 5, 6], "zebrafish": 3, "log": [3, 5], "determin": [3, 4], "allow": [3, 6], "make": [3, 6], "decis": 3, "often": 3, "enhanc": [3, 6], "accuraci": 3, "prefer": [1, 3], "data_sourc": 3, "epimap": 3, "roadmap": 3, "origin": 3, "111": 3, "nih": 3, "2015": 3, "current": [3, 6], "releas": [3, 6], "ha": 3, "expand": 3, "its": 3, "scope": 3, "offer": [3, 5], "comprehens": [1, 3, 4, 5], "833": 3, "tissu": 3, "boix": 3, "2021": 3, "choos": 3, "backward": 3, "compat": 3, "auto": [3, 5], "forc": 3, "without": [1, 3], "evalu": [4, 5], "top": [0, 4, 5, 6], "100": [3, 4, 5, 6], "differenti": [0, 4], "pinpoint": [], "perform": [1, 4, 6], "princip": 4, "compon": 4, "extract": [0, 4, 5, 6], "orthogon": 4, "pattern": [3, 4], "deposit": 4, "level": [4, 5], "bayesian": 4, "criterion": 4, "assess": 4, "search": [4, 7], "interact": 4, "ppi": 4, "network": 4, "string": [4, 6], "databas": [1, 4, 6], "conduct": [4, 6], "direct": [4, 6], "provid": [1, 4, 5, 6, 7], "As": 4, "typic": 4, "line": 4, "ideal": 4, "along": [4, 5], "_gene_clust": 4, "probabl": [4, 6], "being": 4, "assign": [4, 6], "analys": [4, 6], "involv": 4, "multipl": 4, "store": 4, "_go": 4, "signific": [4, 6], "discoveri": 4, "rate": 4, "all": [1, 3, 4, 5, 6], "term": [0, 4, 6], "triageparser_output": [4, 6], "h3k27me3_pc": 4, "pre": 4, "pca_roadmap": 4, "number_of_pca": 4, "10": [0, 3, 4, 5, 6], "number_of_gen": 4, "no_it": 4, "iter": 4, "best": 4, "bic": 4, "em_tol": 4, "converg": 4, "threshold": [4, 6], "expect": 4, "maxim": 4, "em": 4, "gaussianmixtur": 4, "1e": 4, "em_max_it": 4, "maximum": 4, "go_analysi": 4, "ye": 4, "verbos": 4, "max_clust": 4, "gene_ord": 4, "sort": [4, 6], "ascend": 4, "go_threshold": 4, "01": [3, 4, 6], "200": 4, "suit": 5, "tool": 5, "transcriptom": [5, 7], "document": [5, 6], "guid": [5, 7], "correct": 5, "oper": 5, "input": [5, 6, 7], "enabl": [1, 3, 5], "triagegene_demo_human": 5, "header": 5, "dir": 5, "exist": [1, 5], "creat": [1, 5, 6], "jaccard_heatmap_human_test1": 5, "triagegene_demo_mous": 5, "jaccard_heatmap_mouse_test2": 5, "top_no": [0, 5, 6], "convert": 5, "88": 5, "demo_matrix": 5, "jaccard_heatmap_mouse_test3": 5, "test4": 5, "peak_column": [5, 6], "machin": 5, "test5": 5, "6": [5, 6], "test6": 5, "id": [5, 6], "7": [5, 6], "triageparser_demo_genelist": 5, "test7": 5, "serv": 5, "practic": 5, "complex": [1, 5, 6], "procedur": 5, "effect": 5, "project": [1, 5], "sever": 6, "intuit": 6, "graphic": 6, "represent": 6, "easier": 6, "interpret": 6, "three": 6, "comparison": 6, "output_fil": [5, 6], "desir": 6, "usag": [6, 7], "input_matrix": 6, "jaccard_heatmap": 6, "support": 6, "expr_df": 6, "metadata_df": 6, "cell_nam": 6, "highest": 6, "particularli": 6, "cluster_nam": 6, "cluster1": [5, 6], "cluster1_gen": [5, 6], "output_gene_clust": 6, "particular": 6, "color_palett": 6, "custom": 6, "color": 6, "palett": 6, "gradient": 6, "light": 6, "grei": 6, "red": 6, "top_term": 6, "width": 6, "behavior": 6, "height": 6, "heatmap_output": 6, "peak01": [], "size": 6, "6x7": 6, "instruct": [3, 7], "bulk": 7, "instal": 7, "depend": 7, "python": 7, "environ": 7, "setup": 7, "knowledg": [], "descript": 7, "friendli": 7, "getclustergen": [0, 7], "case": 7, "studi": 7, "page": 7, "bw": 2, "vector": [2, 6], "kde": 2, "etc": 2, "classifi": [4, 5], "fdr": 6, "visul": 6, "peak0_gene_clust": 5, "cluster4": 0, "show": [], "highli": 0, "fate": 0, "commit": 0, "peak3_cluster4_gen": 0, "peak3_gene_clust": 0, "sinc": 0, "top_ds_gen": 5, "n": [1, 6], "should": 6, "return": 6, "element": 6, "genesymbol": 6, "top_gen": [0, 5, 6], "topgen": [0, 7], "ad": 6, "v1": 6, "clusterprofil": 1, "enrichplot": 1, "dplyr": 1, "ggplot2": 1, "reshape2": 1, "onli": [1, 3, 6], "annot": [1, 6], "org": [1, 5, 6], "h": [1, 6], "eg": [1, 5, 6], "db": [1, 5, 6], "requirenamespac": 1, "biocmanag": 1, "quietli": 1, "common": [1, 6], "mm": [1, 5, 6], "mice": 1, "rn": [1, 6], "rat": [1, 6], "bioconductor": [1, 6], "biocview": 6, "html": 6, "___orgdb": 6, "pvalu": [3, 5], "p": [3, 5], "z": [3, 5], "compar": [3, 5, 6], "min_comparable_gen": 3, "integ": 3, "least": 3, "percentil": 3, "numer": [3, 6], "proport": 3, "fewer": 3, "than": 3, "dot": [5, 6], "group_1": 5, "go_result": 5, "ds_column": [5, 6], "group1_comparego": 5, "cell_1": 5, "20": [5, 6], "50": [5, 6], "80": 5, "cell1_comparego": 5, "analyt": 6, "symbol": 6, "charact": 6, "ds_valu": 6, "orgdb": 6, "addit": [1, 6], "keyword": 6, "filter": 6, "those": 6, "wnt": 6, "cardiac": 6, "pacemak": 6, "muscl": 6, "cardiomyocyt": 6, "fibroblast": 6, "myofibroblast": 6, "smooth": 6, "color_low": 6, "low": 6, "lightgrei": 6, "color_high": 6, "inch": 6, "addition": 6, "150": 6, "focu": 6, "ds_data": 6, "comparego": [1, 7], "eukaryot": 3, "arthur": 3, "genom": 3, "2014": 3, "ortholog": 3, "though": 3, "were": [1, 3], "public": 3, "valid": 3, "ensembl": 3, "biomart": 3, "retriev": 3, "between": 3, "onlin": 3, "found": 3, "2009": 3, "21": 3, "get": 3, "ll": 1, "There": 1, "main": 1, "conda": 1, "virtual": 1, "pyenv": 1, "outlin": 1, "later": 1, "mai": 1, "relev": 1, "have": 1, "been": 1, "env": 1, "don": 1, "forg": 1, "conflict": 1, "verifi": 1, "command": 1, "use_condaenv": 1, "successfulli": 1, "py_module_avail": 1, "isol": 1, "broader": 1, "featur": 1, "start": 1, "bash": 1, "block": []}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"case": 0, "studi": 0, "instal": 1, "r": [1, 7], "depend": 1, "python": 1, "environ": 1, "setup": 1, "knowledg": [], "base": [], "triag": [1, 5, 7], "packag": [1, 7], "triageclust": [2, 5], "descript": [2, 3, 4], "input": [2, 3, 4], "output": [2, 3, 4], "paramet": [2, 3, 4], "usag": [2, 3, 4], "exampl": [2, 3, 4], "triagegen": [3, 5], "triagepars": [4, 5], "test": 5, "plotjaccard": [5, 6], "bypeak": [5, 6], "plotgo": [5, 6], "user": 6, "friendli": 6, "function": 6, "getclustergen": [5, 6], "welcom": 7, "": 7, "document": 7, "content": 7, "indic": 7, "tabl": 7, "topgen": [5, 6], "addit": [], "comparego": [5, 6]}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 58}, "alltitles": {"TRIAGEcluster": [[2, "triagecluster"]], "Description": [[2, "description"], [4, "description"], [3, "description"]], "Input and Output": [[2, "input-and-output"], [4, "input-and-output"], [3, "input-and-output"]], "Parameters": [[2, "parameters"], [4, "parameters"], [3, "parameters"]], "Usage Examples": [[2, "usage-examples"], [4, "usage-examples"], [3, "usage-examples"]], "TRIAGEparser": [[4, "triageparser"]], "Welcome to TRIAGE R Package\u2019s documentation!": [[7, "welcome-to-triage-r-package-s-documentation"]], "Contents:": [[7, "contents"]], "Indices and tables": [[7, "indices-and-tables"]], "Case Study": [[0, "case-study"]], "Testing TRIAGE": [[5, "testing-triage"]], "Test TRIAGEgene + plotJaccard() + compareGO()": [[5, "test-triagegene-plotjaccard-comparego"]], "Test TRIAGEcluster + byPeak() + topGenes()": [[5, "test-triagecluster-bypeak-topgenes"]], "Test TRIAGEparser + plotGO() + getClusterGenes()": [[5, "test-triageparser-plotgo-getclustergenes"]], "TRIAGEgene": [[3, "triagegene"]], "User-friendly Functions": [[6, "user-friendly-functions"]], "plotJaccard": [[6, "plotjaccard"]], "byPeak": [[6, "bypeak"]], "topGenes": [[6, "topgenes"]], "getClusterGenes": [[6, "getclustergenes"]], "plotGO": [[6, "plotgo"]], "compareGO": [[6, "comparego"]], "Installation": [[1, "installation"]], "R Dependencies": [[1, "r-dependencies"]], "Python Environment Setup": [[1, "python-environment-setup"]], "Installing TRIAGE R Package": [[1, "installing-triage-r-package"]]}, "indexentries": {}})